examples/utils/generic.py

```diff
@@ -91,27 +144,24 @@ def ensure_packages_are_installed(*names):
     if len(names) > 0:
         for name in names:
             if importlib.util.find_spec(name) is None:
-                install_package(name)
+                install_package(name, notebook_environment)
 
     # Install requirements.txt if nothing was passed in
     else:
-        reqs_file = os.path.realpath(os.path.join(__file__, "../../../requirements.txt"))
+        reqs_file = get_requirements_filepath(notebook_environment)
         with open(reqs_file, "r") as reqs:
             for line in reqs:
                 # Ignore Jupyterlab, since the user is probably running it already to view the notebooks
                 if "jupyterlab" in line:
                     pass
                 # Check if packages exist, and install if they don't
                 else:
-                    name, version = line.strip().split("==")
-                    if importlib.util.find_spec(name) is None:
-                        install_package(name, version)
-
-
-ensure_packages_are_installed("tabulate")
-
-# Needs to go here, to ensure it's installed before we import
-from tabulate import tabulate
+                    # Because of how we read the lines, blank lines are represented as '\n'.
+                    # The following condition treats blank lines given as '\n' as well as '==' in comments.
+                    if '==' in line and '#' not in line:
```

Comment:
Let's use a regex, maybe - something like https://stackoverflow.com/questions/25822749/python-regex-for-matching-single-line-and-multi-line-comments?